File: Data Structures & Algorithms/three-integer-sum/submission-3.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        prev = None
        for index in range(len(nums)):
            if prev == nums[index]:
                continue
            prev = nums[index]
            left, right = index + 1, len(nums) - 1
            while left < right:
                threeSum = nums[index] + nums[left] + nums[right]
                if threeSum > 0:
                    right -= 1
                elif threeSum < 0:
                    left += 1
                else:
                    ans.append([nums[index], nums[right], nums[left]])
                    left += 1
                    right -= 1
                    while nums[left] == nums[left - 1] and left < right:
                        left += 1
        return ans

File: Data Structures & Algorithms/valid-binary-search-tree/submission-3.js
class Solution {
	/**
	 * @param {TreeNode} root
	 * @return {boolean}
	 */
	isValidBST(root) {
		if (!root.left && !root.right) return true;
		if (root.left && root.left.val >= root.val) return false;
		if (root.right && root.right.val < root.val) return false;
		if (root.left)
			return this.isValidBST(root.left)
		if (root.right)
			return this.isValidBST(root.right)
	}
}
